iconservice/icon_config.py

```diff
@@ -62,5 +72,177 @@
     ConfigKey.STEP_TRACE_FLAG: False,
     ConfigKey.PRECOMMIT_DATA_LOG_FLAG: False,
     ConfigKey.BACKUP_FILES: BACKUP_FILES,
-    ConfigKey.BLOCK_INVOKE_TIMEOUT: BLOCK_INVOKE_TIMEOUT_S
+    ConfigKey.BLOCK_INVOKE_TIMEOUT: BLOCK_INVOKE_TIMEOUT_S,
+    ConfigKey.TBEARS_MODE: False,
 }
+
+
+def check_config(conf: dict, default_conf: dict) -> bool:
+    checker = ConfigSanityChecker(default_conf)
+    ret: bool = checker.run(conf)
+
+    if len(checker.invalid_keys) > 0:
+        Logger.info(tag=_TAG, msg="===== INVALID CONFIGURATION KEYS =====")
+        for item in checker.invalid_keys:
+            Logger.info(tag=_TAG, msg=f"{item}")
+
+    if len(checker.invalid_values) > 0:
+        Logger.error(tag=_TAG, msg="===== INVALID CONFIGURATION VALUE TYPES ===== ")
+        for item in checker.invalid_values:
+            Logger.error(tag=_TAG, msg=f"{item}")
+
+    return ret
+
+
+def args_to_dict(args) -> dict:
+    """Convert args to dict
+    :param args: command line arguments
+    :return: conf in dict containing command line arguments
+    """
+    keys_to_remove = {"command", "foreground", "config"}
+    args_dict = vars(args)
+    ret = {}
+
+    for key, value in args_dict.items():
+        if key not in keys_to_remove and value is not None:
+            ret[key] = value
+
+    return ret
+
+
+class ConfigSanityChecker(object):
+    """Sanity check for iconservice configuration file
+    """
+
+    class Item(object):
+        def __init__(self,
+                     key: str,
+                     value: Union[int, str, bool, Dict],
+                     value_type: Union[None, dict, Type[int], Type[str], Type[bool], Type[float]]):
```

Comment:
How about using `Generic` type hint with constrained?
https://docs.python.org/ko/3/library/typing.html#user-defined-generic-types
```py
S = TypeVar('S', int, str)  # Must be int or str
```

Reply:
Good idea!!!

Reply:
Fixed it with type alias.